### PR TITLE
Add descriptions to ovms.h

### PR DESCRIPTION
**Pull request by atobiszei**
JIRA:CVS-98964

### CODE REVIEW

**Review by rasapala — APPROVED** [review]
LGTM

File: src/ovms.h [hunk 1]
```diff
@@ -27,7 +27,10 @@ typedef struct OVMS_Status_ OVMS_Status;
 typedef struct OVMS_ServerSettings_ OVMS_ServerSettings;
 typedef struct OVMS_ModelsSettings_ OVMS_ModelsSettings;
 
-// TODO reuse this in precision.hpp
+// OVMS_DataType
+//
+// Tensor and parameter data types recognized by OVMS
```
Comment (dkalinowski) [inline src/ovms.h:32]: Do we recognize other parameter types than int and string?

Comment (atobiszei) [reply src/ovms.h:32]: Right now we don't use parameters at all. THey are placeholder for Stateful support & reusing C-API inference by both TFS & KFS

Comment (dkalinowski) [reply src/ovms.h:32]: Then the user can feel misguided that we will end up supporting all the precisions for parameters

File: src/ovms.h [hunk 2]
```diff
@@ -91,63 +115,147 @@ OVMS_Status* OVMS_StatusGetDetails(OVMS_Status* status,
...
 OVMS_Status* OVMS_ServerSettingsNew(OVMS_ServerSettings** settings);
-// Deallocates server settings memory for given ptr
+
+// Deallocates server settings object for given ptr
+//
+// \param settings The settings object to be removed
 void OVMS_ServerSettingsDelete(OVMS_ServerSettings* settings);
 
-// --port
+// Set the gRPC port of starting OVMS. Equivalent of using --port parameter from OVMS CLI
```
Comment (dkalinowski) [inline src/ovms.h:128]: Add that default is 9178?

File: src/ovms.h [hunk 3]
```diff
@@ -171,37 +290,143 @@ OVMS_Status* OVMS_ModelsSettingsSetConfigPath(OVMS_ModelsSettings* settings,
...
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_ServerNew(OVMS_Server** server);
+
 // Deallocates server memory for given ptr
+//
+// \param server The server object to be removed
 void OVMS_ServerDelete(OVMS_Server* server);
 
 // Start with configuration file config.json
 // Return error if already started
```
Comment (dkalinowski) [inline src/ovms.h:304]: Return error if already started or any other loading error (configuration error)

File: src/ovms.h [hunk 4]
```diff
@@ -171,37 +290,143 @@ OVMS_Status* OVMS_ModelsSettingsSetConfigPath(OVMS_ModelsSettings* settings,
...
 // Unload all and cleanup
 
 // OVMS_InferenceRequest
+//
+// Create new inference request object
+//
+// \param request The request object to be created
+// \param server The server object
+// \param servableName The name of the servable to be used
+// \param servableVersion The version of the servable to be used
```
Comment (dkalinowski) [inline src/ovms.h:322]: Add info about optional `0` as "any" version?

File: src/ovms.h [hunk 5]
```diff
@@ -171,37 +290,143 @@ OVMS_Status* OVMS_ModelsSettingsSetConfigPath(OVMS_ModelsSettings* settings,
...
+// \param dimCount The number of dimensions of the shape
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_InferenceRequestAddInput(OVMS_InferenceRequest* request, const char* inputName, OVMS_DataType datatype, const uint64_t* shape, uint32_t dimCount);
 
-// ownership of data needs to be maintained during inference
-OVMS_Status* OVMS_InferenceRequestInputSetData(OVMS_InferenceRequest* request, const char* inputName, const void* data, size_t bufferSize, OVMS_BufferType bufferType, uint32_t deviceId);
+// Set the data of the input buffer. Ownership of data needs to be maintained during inference
+//
+// \param request The request object
+// \param inputName The name of the input with data to be removed
```
Comment (dkalinowski) [inline src/ovms.h:340]: To be removed or added?

File: src/ovms.h [hunk 6]
```diff
@@ -171,37 +290,143 @@ OVMS_Status* OVMS_ModelsSettingsSetConfigPath(OVMS_ModelsSettings* settings,
...
 OVMS_Status* OVMS_InferenceResponseGetOutputCount(OVMS_InferenceResponse* response, uint32_t* count);
-OVMS_Status* OVMS_InferenceResponseGetOutput(OVMS_InferenceResponse* response, uint32_t id, const char** name, OVMS_DataType* datatype, const uint64_t** shape, uint32_t* dimCount, const void** data, size_t* bytesize, OVMS_BufferType* bufferType, uint32_t* deviceId);
+
+// Get all information about an output from the response.
+//
+// \param response The response object
+// \param id The id of the output
+// \param name The name of the output
+// \param datatype The data type of the output
+// \param shape The shape of the output
```
Comment (dkalinowski) [inline src/ovms.h:395]: Should we distinguish `**` from `*` in the description? Some `in`/`out` ? So end user is sure what is user input and what is user output:
Example: response/id is input
all the rest is output in this case

As you can see user cannot deduce from **/*, for example datatype

Suggestion
```
// \param response [in] The response object
// \param id [in] The id of the output
// \param name [out] The name of the output
// \param datatype [out] The data type of the output
...
```

Comment (atobiszei) [reply src/ovms.h:395]: The name of the function suggest that we are getting something. See also the first comment about this function,

Comment (dkalinowski) [reply src/ovms.h:396]: Yes, we get something, but not all params are for "getting something" - first 2 params are for providing something

File: src/ovms.h (hunk 1 above)
Comment (atobiszei) [reply src/ovms.h]: It's rather question of which parameters we actually support. For now candidates are sequence_id, sequence_control_input, & one parameter for KFS informing where to put the  buffer content. So right now we limit the support of parameters by not supporting any in preview (we are just not using those), and we can whitelist parameter and support them one by one as we have to.

File: src/ovms.h (hunk 6 above)
Comment (atobiszei) [reply src/ovms.h:396]: Added clearer description in the comment

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**